source/common/http/conn_manager_impl.cc

```diff
@@ -1781,7 +1792,10 @@ ConnectionManagerImpl::ActiveStream::route(const Router::RouteCallback& cb) {
   if (cached_route_.has_value()) {
     return cached_route_.value();
   }
-  refreshCachedRoute(cb);
+
+  const auto route_config = getRouteConfig();
+  refreshCachedRoute(route_config.first, cb);
+
```

Comment:
comment: Because we will not cache the entire route config in the active request, we need get the route config from the hcm again when we refresh the route.
This will result in a minor behavior change. That's say if a request is comming and a initial route is selected. And then, the route config is updated by the RDS. And then the cached route is clear and refreshed by one filter in the filter chain. In the previous impl, the old version route config will be used to refresh the route and in the new impl the latest route config will be used to refresh the route.

Reply:
The target services/clusters is managed by the cm and I think the route change will not effects it.

Except the behavior change, this refresh should be safe.

But anyway, I think your idea to clear the cached route table explicitly is great, so I will update this patch in next days.